server/controllers/profileController.js

```diff
@@ -0,0 +1,67 @@
+const Profile = require("../models/ProfileModel");
+
+// create new profile using given arguments
+
+exports.createProfile = async (req, res, next) => {
+  try {
+    const { firstName, lasttName, phone, address, gender, description } =
+      req.body;
+    let data = {
+      userId: req.params.id,
+      firstName: firstName.trim().toString(),
+      lasttName: lasttName.trim().toString(),
+      phone: phone.trim().toString(),
+      address: address.trim().toString(),
+      gender: gender.trim().toString(),
+      description: description.trim().toString(),
```

Comment:
Validate these fields before invoking any methods on them. For example, if `firstName` is `undefined`,
`firstName: firstName.trim().toString()` will raise an exception.